### PR TITLE
Clarify the purpose of RichTextLabel text highlight padding

### LINKED ISSUE #79534: RichTextLabel text padding options do not affect highlighted text
### Godot version

v4.1.stable.official [970459615]

### System information

Godot v4.1.stable - Windows 10.0.22621 - Vulkan (Forward+) - dedicated NVIDIA GeForce RTX 3080 (NVIDIA; 31.0.15.3598) - 11th Gen Intel(R) Core(TM) i5-11400 @ 2.60GHz (12 Threads)

### Issue description

Based on #63590 and the [documentation](https://docs.godotengine.org/en/stable/classes/class_richtextlabel.html#class-richtextlabel-theme-constant-text-highlight-v-padding) for text_highlight_h_padding and text_highlight_v_padding, changes to these values should affect the width and height of the color box created from highlighting text, but this is not the case as the size of the color box remains the same.

Note that these values do affect the size of the color box created by the [bgcolor] bbcode.

image
This image depicts a RichTextLabel with both text_highlight_h_padding and text_highlight_v_padding set to 10, a magenta bgcolor, and the text being highlighted.

### Steps to reproduce

Add a RichTextLabel and enable text selection.
Change the values of  text_highlight_h_padding or text_highlight_v_padding to something non-zero.
Run the project and highlight the text.

### Minimal reproduction project

N/A

### PR DESCRIPTION
**Pull request by Calinou**
- This closes https://github.com/godotengine/godot/issues/79534.

### CODE REVIEW

File: doc/classes/RichTextLabel.xml [hunk 1]
```diff
@@ -706,10 +706,10 @@
 			The vertical separation of elements in a table.
 		</theme_item>
 		<theme_item name="text_highlight_h_padding" data_type="constant" type="int" default="3">
-			The horizontal padding around a highlighting and background color box.
+			The horizontal padding around boxes drawn by the [code][fgcolor][/code] and [code][bgcolor][/code] tags. This does not affect the appearance of text selection.
```
Comment (YuriSizov) [inline doc/classes/RichTextLabel.xml:709]:
```suggestion
			The horizontal padding around boxes drawn by the [code skip-lint][fgcolor][/code] and [code skip-lint][bgcolor][/code] tags. This does not affect the appearance of text selection.
```

File: doc/classes/RichTextLabel.xml [hunk 2]
```diff
@@ -706,10 +706,10 @@
...
 		</theme_item>
 		<theme_item name="text_highlight_v_padding" data_type="constant" type="int" default="3">
-			The vertical padding around a highlighting and background color box.
+			The vertical padding around boxes drawn by the [code][fgcolor][/code] and [code][bgcolor][/code] tags. This does not affect the appearance of text selection.
```
Comment (YuriSizov) [inline doc/classes/RichTextLabel.xml:712]:
```suggestion
			The vertical padding around boxes drawn by the [code skip-lint][fgcolor][/code] and [code skip-lint][bgcolor][/code] tags. This does not affect the appearance of text selection.
```

**Review by YuriSizov — APPROVED** [review]
Needs some updates to make sure it passes CI, but looks good otherwise.

Note, that these updates would make it unpickable for 4.1.

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**